src/state/application/updater.ts

```diff
@@ -20,6 +22,40 @@ function useQueryCacheInvalidator() {
   }, [chainId, dispatch])
 }
 
+const NETWORK_HEALTH_CHECK_MS = ms`15s`
+const DEFAULT_MS_BEFORE_WARNING_WAIT = ms`10m`
+interface UseBlockWarningTimerArgs {
+  chainId: number | undefined
+  dispatch: (action: any) => void
+  msSinceLastBlock: number
+  setMsSinceLastBlock: (n: number) => void
+}
+
+function useBlockWarningTimer({ chainId, dispatch, msSinceLastBlock, setMsSinceLastBlock }: UseBlockWarningTimerArgs) {
+  const chainConnectivityWarningActive = useAppSelector((state) => state.application.chainConnectivityWarning)
+  const timeout = useRef<NodeJS.Timeout>()
+  useEffect(() => {
+    const waitMsBeforeWarning =
+      (chainId ? CHAIN_INFO[chainId]?.blockWaitMsBeforeWarning : DEFAULT_MS_BEFORE_WARNING_WAIT) ||
+      DEFAULT_MS_BEFORE_WARNING_WAIT
+
+    timeout.current = setTimeout(() => {
+      setMsSinceLastBlock(NETWORK_HEALTH_CHECK_MS + msSinceLastBlock)
+      if (msSinceLastBlock > waitMsBeforeWarning) {
+        dispatch(setChainConnectivityWarning({ warn: true }))
```

Comment:
why are you passing the dispatch as an arg? Isn't it always the same `appDispatch`?